test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,213 @@
+                                       fm_index_seqan2_t,
+                                       std::conditional_t<std::same_as<index_tag_t, bi_fm_index_seqan2>,
+                                                          bi_fm_index_seqan2_t,
+                                                          std::conditional_t<std::same_as<index_tag_t, fm_index_seqan3>,
+                                                                             fm_index_seqan3_t,
+                                                                             bi_fm_index_seqan3_t>>>;
+#else
+    using index_t = std::conditional_t<std::same_as<index_tag_t, fm_index_seqan3>,
+                                       fm_index_seqan3_t,
+                                       bi_fm_index_seqan3_t>;
+#endif
+};
+
+template <typename index_tag_t, typename rng_t>
+void index_benchmark(benchmark::State & state)
+{
+    sequence_generator<index_tag_t, rng_t> generator(state.range(0), state.range(1));
+    for (auto _ : state)
+    {
+#if SEQAN3_HAS_SEQAN2
+        if constexpr (std::same_as<index_tag_t, fm_index_seqan2> || std::same_as<index_tag_t, bi_fm_index_seqan2>)
+        {
+            typename sequence_generator<index_tag_t, rng_t>::index_t index{generator.sequence_seqan2};
+            seqan::indexCreate(index, seqan::FibreSALF());
+        }
+        else
+#endif // SEQAN3_HAS_SEQAN2
+            typename sequence_generator<index_tag_t, rng_t>::index_t index{generator.sequence_seqan3};
+    }
+}
+
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan3, std::vector<seqan3::dna4>)->Apply(arguments);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan2, std::vector<seqan3::dna4>)->Apply(arguments);
+#endif // SEQAN3_HAS_SEQAN2
+
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan3, std::vector<seqan3::aa27>)->Apply(arguments);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan2, std::vector<seqan3::aa27>)->Apply(arguments);
+#endif // SEQAN3_HAS_SEQAN2
+
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan3, std::string)->Apply(arguments);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(index_benchmark, fm_index_seqan2, std::string)->Apply(arguments);
+#endif // SEQAN3_HAS_SEQAN2
+
+BENCHMARK_TEMPLATE(index_benchmark, bi_fm_index_seqan3, std::vector<seqan3::dna4>)->Apply(arguments);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(index_benchmark, bi_fm_index_seqan2, std::vector<seqan3::dna4>)->Apply(arguments);
+#endif // SEQAN3_HAS_SEQAN2
+
+BENCHMARK_TEMPLATE(index_benchmark, bi_fm_index_seqan3, std::vector<seqan3::aa27>)->Apply(arguments);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(index_benchmark, bi_fm_index_seqan2, std::vector<seqan3::aa27>)->Apply(arguments);
+#endif // SEQAN3_HAS_SEQAN2
+
+BENCHMARK_TEMPLATE(index_benchmark, bi_fm_index_seqan3, std::string)->Apply(arguments);
+#if SEQAN3_HAS_SEQAN2
+BENCHMARK_TEMPLATE(index_benchmark, bi_fm_index_seqan2, std::string)->Apply(arguments);
+#endif // SEQAN3_HAS_SEQAN2
```

Comment:
Why don't you group all the seqan2 benchmarks in one `#if` block and after that have all seqan3 benchmarks?

Reply:
I wanted to have the equivalent benchmarks in seqan2 and seqan3 next to each other, rather than having a long block of seqan3 related benchmarks and then a long block of seqan2 related benchmarks...